### PR TITLE
prov/util, shm: fix util srx to adhere to peer api (setting msg_size and flags correctly)

### PR DESCRIPTION
**Pull request by aingerson**
fix util_srx implementation to properly set msg_size and flags and remove setting in shm

### CODE REVIEW

**Comment by aingerson** [comment]
@shijin-aws @amirshehataornl Any thoughts or concerns?

**Review by j-xiong — APPROVED** [review]
LGTM

**Review by shijin-aws — APPROVED** [review]

File: prov/util/src/util_srx.c [hunk 1]
```diff
@@ -496,6 +501,33 @@ static struct util_rx_entry *util_search_unexp_msg(struct util_srx_ctx *srx,
...
+static bool util_unexp_mrecv(struct util_srx_ctx *srx,
+			     struct util_rx_entry *mrecv_entry,
+			     struct util_rx_entry *rx_entry)
+{
+	mrecv_entry->multi_recv_ref++;
+	rx_entry->peer_entry.owner_context = mrecv_entry;
+
+	rx_entry->peer_entry.iov[0].iov_base =
+			mrecv_entry->peer_entry.iov->iov_base;
+	rx_entry->peer_entry.iov->iov_len =
```
Comment (shijin-aws) [inline prov/util/src/util_srx.c:513]: It's a bit strange that we use `iov[0]` for iov_base while `iov->` for iov_len. Is it enforced that multi_recv post has to be 1 count?

Comment (j-xiong) [reply prov/util/src/util_srx.c:513]: Yes, current util code only supports iov_count == 1 for multi_recv. See the assertion in `util_generic_mrecv`：
```
	assert(flags & FI_MULTI_RECV && iov_count == 1);
```

**Pull request merged**